randomsub.go

```diff
@@ -119,6 +123,14 @@ func (rs *RandomSubRouter) Publish(msg *Message) {
 	}
 
 	if len(rspeers) > RandomSubD {
+		target := RandomSubD
+		sqrt := int(math.Ceil(math.Sqrt(float64(rs.size))))
+		if sqrt > target {
+			target = sqrt
+		}
+		if target > len(rspeers) {
+			target = len(rspeers)
+		}
 		xpeers := peerMapToList(rspeers)
 		shufflePeers(xpeers)
 		xpeers = xpeers[:RandomSubD]
```

Comment:
It goes through all the work of calculating target, but it's not actually used - should it replace the use of `RandomSubD` here?

Reply:
ooops, that's a bug! thanks for spotting.

Reply:
Care for a patch?